src/mo_frontend/typing.ml

```diff
@@ -135,6 +135,14 @@ let error_in modes env at code fmt =
 
 let plural cs = if T.ConSet.cardinal cs = 1 then "" else "s"
 
+let warn_lossy_bind_type env at bind t1 t2 =
+  let common = T.lub t1 t2 in
+  if T.(not (eq t1 common || eq common t2)) then
+    warn env at "M0185" "inconsistent type for alternative pattern variables %s given%a%a"
```

Comment:
```suggestion
    warn env at "M0185" "pattern variabels %a has type%a\nbecause its types in the pattern alternatives are inconsistent,\n type in left pattern is%a\nright type in right pattern is%a"
```

Reply:
```suggestion
    warn env at "M0185" "pattern variable %a has larger type%a\nbecause its types in the pattern alternatives are unrelated smaller types: \ntype in left pattern is%a\ntype in right pattern is%a"
```
(if you aren't using consistency, just subtyping, maybe this is more appropriate).

Reply:
Now you know why I was hesitant to come up with a text :-)